rush-exec/src/errors.rs

```diff
@@ -1,7 +1,70 @@
+use std::io::ErrorKind;
 use std::path::PathBuf;
+use std::{fmt, io};
 
+use anyhow::anyhow;
 use thiserror::Error;
 
+/// This is a wrapper for io::Error to add more context than the default Display.
+/// It should not be used directly. Use an internal error instead.
+#[derive(Error, Debug)]
+pub struct IoError {
+    #[from]
+    source: io::Error,
+}
+
+impl fmt::Display for IoError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.source)?;
+        if let Some(e) = self.source.get_ref() {
+            if let Some(e) = e.source() {
+                write!(f, " because: {}", e)?;
+            }
+        }
+        Ok(())
+    }
+}
+
+pub enum FileContext {
```

Comment:
I think that contexts should all be contained in a single struct unless the variants count gets extremely out of hand.